pandas/tseries/offsets.py

```diff
@@ -182,6 +182,29 @@ def __add__(date):
     date + BDay(0) == BDay.rollforward(date)
 
     Since 0 is a bit weird, we suggest avoiding its use.
+
+    Parameters
```

Comment:
correct, can you separate this into 2 subsections and make clear the different behavior (via an example)

```
In [1]: t = pd.Timestamp('20170101 09:10:11')

# operate with the value
In [3]: t + DateOffset(years=2018)
Out[3]: datetime.datetime(4035, 1, 1, 9, 10, 11)

In [5]: t + DateOffset(seconds=1)
Out[5]: Timestamp('2017-01-01 09:10:12')

# set the value
In [6]: t + DateOffset(second=1)
Out[6]: Timestamp('2017-01-01 09:10:01')

In [8]: t + DateOffset(month=1)
Out[8]: Timestamp('2017-01-01 09:10:11')
```

Reply:
there is an issue to revisit this (IOW we should have 2 different offsets for this different behaviors)

Reply:
Sure.  The docstring is already pretty big and likely to get much bigger.  Is there somewhere else part of it should go?

>there is an issue to revisit this (IOW we should have 2 different offsets for this different behaviors)

Yah, IIRC there was a suggestion to have `__new__` dispatch some special cases to `Tick`, `MonthOffset`, etc.  I'm planning to take a swing at that after the immutabilization since `__new__` and cython are tricky.

Reply:
don't make it long, make it wide instead (its a bit non-standard ok)

```
n : int, default 1
normalize : bool, default False

Parameters that ADD to the offset
----------------------------------
years, months, ....

Parameters that REPLACE the offset value
-----------
year, month

```